app/src/main/java/com/github/bgabriel998/softwaredevproject/Button1Activity.java

```diff
@@ -136,12 +151,141 @@ protected void onResume() {
     }
 
     /**
-     *  Unbind and shutdown camera before exiting camera and stop the compass
+     * Unbind and shutdown camera before exiting camera and stop the compass
      */
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        cameraPreview.destroy();
+        //cameraPreview.destroy();
         compass.stop();
     }
+
+    /**
+     * Handle orientation changes
+     *
+     * @param newConfig new device configuration
+     */
+    @Override
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        ImageButton takePictureButton = findViewById(R.id.takePicture);
+
+        ConstraintLayout.LayoutParams params = (ConstraintLayout.LayoutParams) takePictureButton.getLayoutParams();
+
+        //Change constraints of the ImageButton depending on the orientation
+        if (newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            params.rightToRight = R.id.cameraLayout;
+            params.topToTop = R.id.cameraLayout;
+            params.bottomToBottom = R.id.cameraLayout;
+            params.endToEnd = -1;
+            params.startToStart = -1;
+            params.rightMargin = 10;
+
+        } else if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
+            params.rightToRight = -1;
+            params.topToTop = -1;
+            params.endToEnd = R.id.cameraLayout;
+            params.startToStart = R.id.cameraLayout;
+            params.bottomToBottom = R.id.compass;
+            params.rightMargin = 0;
+        }
+
+        //Change the constraints of the button
+        takePictureButton.setLayoutParams(params);
+    }
```

Comment:
Is this function used to change the placement and rotation for layouts (like camera button etc.) Because we can create 2 layout xml files to solve that, one for portrait and one for landscape. 
If it is only that the function does, then maybe we should add an TODO regarding to remove this function when the camera is added to the UI.
```suggestion
    /**
     * Handle orientation changes
     * TODO, remove function and create seperate layout xml files for portrait and landscape
     * @param newConfig new device configuration
     */
    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);

        ImageButton takePictureButton = findViewById(R.id.takePicture);

        ConstraintLayout.LayoutParams params = (ConstraintLayout.LayoutParams) takePictureButton.getLayoutParams();

        //Change constraints of the ImageButton depending on the orientation
        if (newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
            params.rightToRight = R.id.cameraLayout;
            params.topToTop = R.id.cameraLayout;
            params.bottomToBottom = R.id.cameraLayout;
            params.endToEnd = -1;
            params.startToStart = -1;
            params.rightMargin = 10;

        } else if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
            params.rightToRight = -1;
            params.topToTop = -1;
            params.endToEnd = R.id.cameraLayout;
            params.startToStart = R.id.cameraLayout;
            params.bottomToBottom = R.id.compass;
            params.rightMargin = 0;
        }

        //Change the constraints of the button
        takePictureButton.setLayoutParams(params);
    }
```

Reply:
For the layout, I just added a new .XML file where I handle the orientation change of the button